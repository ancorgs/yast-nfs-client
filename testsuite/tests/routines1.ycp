/**
 * Module:
 *   NFS client configuration
 *
 * Summary:
 *   Routines testuite
 *
 * Authors:
 *   Martin Vidner <mvidner@suse.cz>
 *
 * $Id$
 */
{
    include "testsuite.ycp";
    include "nfs/routines.ycp";

    list nfs_entries = [
	$[
	    "file":"/home",
	    "freq":0,
	    "mntops":"defaults",
	    "passno":0,
	    "spec":"foo.bar.com:/home",
	    "vfstype":"nfs"
	 ],
	$[
	    "file":"/var/spool/mail",
	    "freq":0,
	    "mntops":"defaults",
	    "passno":0,
	    "spec":"foo.bar.com:/var/spool/mail",
	    "vfstype":"nfs"
	 ]
	];

    DUMP ("FstabTableItems");
    TEST (``(FstabTableItems (nfs_entries)), [], nil);

    DUMP ("SpecToServPath");
    TEST (``(SpecToServPath ("big.foo.com:/share/data")), [], nil);
    TEST (``(SpecToServPath ("only.server.com:")), [], nil);
    TEST (``(SpecToServPath ("nocolon.only.server.com")), [], nil);
    TEST (``(SpecToServPath (":/only/path")), [], nil);
    TEST (``(SpecToServPath ("/nocolon/only/path")), [], nil);
    TEST (``(SpecToServPath ("two:colons:used")), [], nil);
    TEST (``(SpecToServPath ("")), [], nil);

    DUMP ("check_options");
    TEST (``(check_options ("")), [], nil);
    TEST (``(check_options ("defaults")), [], nil);
    TEST (``(check_options ("nolock,bg")), [], nil);
    TEST (``(check_options ("nolock,nobg")), [], nil);
    TEST (``(check_options ("nolock,rsize=8192")), [], nil);
    TEST (``(check_options ("nolock, bg")), [], nil);
    TEST (``(check_options ("nolock,unknownoption")), [], nil);
    TEST (``(check_options ("nolock,unknownassignment=true")), [], nil);
    TEST (``(check_options ("nolock,rsize=")), [], nil);
    TEST (``(check_options ("nolock,two=equal=signs")), [], nil);

    DUMP ("StripExtraSlash");
    TEST (``(StripExtraSlash ("")), [], nil);
    TEST (``(StripExtraSlash ("/")), [], nil);
    TEST (``(StripExtraSlash ("/normal/path")), [], nil);
    TEST (``(StripExtraSlash ("/trailing/slash/")), [], nil);
}
