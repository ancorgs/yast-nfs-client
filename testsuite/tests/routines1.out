Dump	FstabTableItems
Return	[`item (`id (0), "foo.bar.com ", "/home ", "/home ", "defaults "), `item (`id (1), "foo.bar.com ", "/var/spool/mail ", "/var/spool/mail ", "defaults ")]
Dump	SpecToServPath
Return	`couple ("big.foo.com", "/share/data")
Return	`couple ("only.server.com", "")
Return	`couple ("", "nocolon.only.server.com")
Return	`couple ("", "/only/path")
Return	`couple ("", "/nocolon/only/path")
Return	`couple ("two", "colons:used")
Return	`couple ("", "")
Dump	check_options
Return	Empty options string not allowed!
Return	
Return	
Return	
Return	
Return	Wrong option format:  bg
Return	Wrong option format: unknownoption
Return	Wrong option: unknownassignment
Return	Empty value for option: rsize
Return	Wrong option format: two=equal=signs
Dump	StripExtraSlash
Return	
Return	/
Return	/normal/path
Return	/trailing/slash
